src/components/app/index.tsx

```diff
@@ -157,7 +207,12 @@ export default class App extends Component<Props, State> {
           </div>
         )}
         {images.map((image, index: number) => (
-          <span class={index ? style.rightLabel : style.leftLabel}>{ENCODER_NAMES[image.type]}</span>
+          <span class={index ? style.rightLabel : style.leftLabel}>
+            <span>{ENCODER_NAMES[image.type]}</span>
+            { (image.data !== undefined) ? (
+              (<a href={image.dataUrl} download={image.type +'.'+image.extension}> 🔻 </a>)
```

Comment:
The file name should be the name of the original file with the extension changed. Although, I'm happy for this to land as long as an issue is created to improve this.